lib/cictl/indexer.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+require "socket"
+require "traject"
+
+require_relative "common"
+
+module CICTL
+  class Indexer
+    include Common
+    DEFAULT_READER_NAME = "jsonl"
+    DEFAULT_WRITER_NAME = "localhost"
+    COLLECTION_MAP_FILE = "collection_code_to_original_from.yaml"
+
+    def initialize(reader: nil, writer: nil)
+      @reader = reader
+      @writer = writer
+      config_paths = [reader_path, writer_path]
+      %w[common common_ht ht subjects].each do |conf|
+        config_paths << File.join(home, "indexers", conf + ".rb")
+      end
+      @indexer = Traject::Indexer::MarcIndexer.new(logger: logger) do |ind|
+        config_paths.each { |config_path| load_config_file(config_path) }
+      end
+    end
+
+    def run(marcfile)
+      # Note: hostname will likely be gibberish under Docker
+      logger.info "Working on #{Socket.gethostname} in #{home}"
```

Comment:
Preference for native methods plus efficiency just in case Ruby is using a C API instead of fork/exec. The bigger question for me is, do we need to log this info at all?

Reply:
I honestly don't know. Right now it's always going to be beeftea-2, and once we get to k8s the hostnames don't matter at all, right? So probably not.